src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -4240,7 +4240,8 @@ void InstanceKlass::log_to_classlist(const ClassFileStream* stream) const {
         // For the boot and platform class loaders, skip classes that are not found in the
         // java runtime image, such as those found in the --patch-module entries.
         // These classes can't be loaded from the archive during runtime.
-        if (!stream->from_boot_loader_modules_image() && strncmp(stream->source(), "jrt:", 4) != 0) {
+        if (!stream->from_boot_loader_modules_image() &&
+            (stream->source() == NULL || strncmp(stream->source(), "jrt:", 4) != 0)) {
```

Comment:
I think the following block on line 4256 should also be changed
```
    if (skip && stream->source() != NULL) {
      tty->print_cr("skip writing class %s from source %s to classlist file",
                    name()->as_C_string(), stream->source());
    } else {
```

Reply:
In fact, the printout (no change) like:
skip writing class java/lang/invoke/BoundMethodHandle$Species_LL from source (null) to classlist file
skip writing class java/lang/invoke/BoundMethodHandle$Species_LLL from source (null) to classlist file
skip writing class java/lang/invoke/BoundMethodHandle$Species_LLLL from source (null) to classlist file
So C++ can handle the NULL arg well in printf. I am not sure if all platforms handle this way, so will change to "(null)":
stream->source() != NULL ? stream->source() : "(null)"